compiler/rustc_ty_utils/src/instance.rs

```diff
@@ -90,13 +92,30 @@ fn resolve_instance_raw<'tcx>(
             let ty = args.type_at(0);
             ty::InstanceKind::AsyncDropGlue(def_id, ty)
         } else {
-            debug!(" => free item");
+            debug!(" => free or inherent item");
             ty::InstanceKind::Item(def_id)
         };
 
         Ok(Some(Instance { def, args }))
     };
-    debug!("resolve_instance: result={:?}", result);
+    if let Ok(Some(Instance { def: ty::InstanceKind::Item(def_id), args })) = result
+        && matches!(tcx.def_kind(def_id), DefKind::Const | DefKind::AssocConst)
+    {
```

Comment:
So it turns out that not all CTFE entry points actually call `try_resolve`, some of them just manually construct an instance for a const item. Not sure what to do about that long term :thinking:

For now though you can go through all the entry points in `rustc_middle/src/mir/interpret/queries` and update everything to use `Instance::expect_resolve` instead of `Instance::new_raw` which should atleast solve any ICEs in the short term until we figure out how to properly handle this